File: 0792-binary-search/0792-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:

        start=0  #initiaizing start point
        end=len(nums)-1  #initializing end point

        while start<=end:
            mid=start+(end-start)//2   #calculating mid point
            if nums[mid]==target:      #BINGO!
                return mid             #return mid index
            elif nums[mid]>target:     #target<mid, go left
                end=mid-1              # shift the end pointer
            elif nums[mid]<target:     #target>mid, go right
                start=mid+1            # shift the start pointer
        return -1                       #NOT Found
        

        

File: 0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {

       int start=0;
       int end=nums.length-1;

        while(start<=end){
            int mid=start+(end-start)/2; //calculate mid point
            if (nums[mid]==target)
            {     
                return mid;   //return mid index
            }          
            else if (nums[mid]>target)  //target<mid, go left
            {   
                end=mid-1;     // shift the end pointer
            }         
            else if (nums[mid]<target) //target>mid, go right
            {    
                start=mid+1;     // shift the start pointer  
            }
        }
        return -1;
 
        
    }
}